scripts/make_ref_speeds.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+import argparse
+import random
+import math
+import os
+import errno
+import sys
+import logging as log
+
+try:
+  import speedtile_pb2
+except ImportError:
+  print 'You need to generate protobuffer source via: protoc --python_out . --proto_path ../proto ../proto/*.proto'
+  sys.exit(1)
+
+#wget https://s3.amazonaws.com/speed-extracts/2017/0/0/002/415.spd.*.gz
+
+###############################################################################
+### tile 2415
+### segment0{refspdlist[hour0:avgspd, hour1:avgspd, hour2:avgspd, etc]} -> sort lo to hi, index in to get refspd20:int, refspd40:int, refspd60:int, refspd80:int
+### segment1{refspdlist[hour0:avgspd, hour1:avgspd, hour2:avgspd, etc]}
+# get the avg speeds for a given segment for each hour (168 hr/day) over 52 weeks
+
+def createAvgSpeedList(fileNameList):
+  #each segment has its own list of speeds, we dont know how many segments for the avg speed list to start with
+  speedListPerSegment = []
+
+  #need to loop thru all of the speed tiles for a given tile id
+  for fileName in fileNameList:
+    #lets load the protobuf speed tile
+    spdtile = speedtile_pb2.SpeedTile()
+    with open(fileName, 'rb') as f:
+      spdtile.ParseFromString(f.read())
+
+    #we now need to retrieve all of the speeds for each segment in this tile
+    for subtile in spdtile.subtiles:
+
+      #make sure that there are enough lists in the list for all segments
+      missing = subtile.totalSegments - len(speedListPerSegment)
+      if missing > 0:
+        speedListPerSegment.extend([ [] for i in range(0, missing) ])
+      #print 'total # created in speedListPerSegment ' + str(len(speedListPerSegment))
+
+      for i, speed in enumerate(subtile.speeds):
+        if speed and speed > 0:
+          speedListPerSegment.insert(subtile.startSegmentIndex + i, speed)
```

Comment:
I believe that this is still not correct

Reply:
Yeah it is wrong. It should not be insert it should be index by segment and append the speed to the list of that segment: 

SpeedListPerSegment[subtile.startSegmentIndex + i].append(speed)